src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/RazorDocumentMappingService.cs

```diff
@@ -22,21 +23,17 @@
 
 namespace Microsoft.AspNetCore.Razor.LanguageServer;
 
-internal sealed class RazorDocumentMappingService : IRazorDocumentMappingService
-{
-    private readonly FilePathService _documentFilePathService;
-    private readonly IDocumentContextFactory _documentContextFactory;
-    private readonly ILogger _logger;
-
-    public RazorDocumentMappingService(
+[Export(typeof(IRazorDocumentMappingService)), Shared]
+[method: ImportingConstructor]
+internal sealed class RazorDocumentMappingService(
         FilePathService filePathService,
         IDocumentContextFactory documentContextFactory,
         IRazorLoggerFactory loggerFactory)
-    {
-        _documentFilePathService = filePathService ?? throw new ArgumentNullException(nameof(filePathService));
-        _documentContextFactory = documentContextFactory ?? throw new ArgumentNullException(nameof(documentContextFactory));
-        _logger = loggerFactory?.CreateLogger<RazorDocumentMappingService>() ?? throw new ArgumentNullException(nameof(loggerFactory));
-    }
+         : IRazorDocumentMappingService
+{
+    private readonly FilePathService _documentFilePathService = filePathService ?? throw new ArgumentNullException(nameof(filePathService));
+    private readonly IDocumentContextFactory _documentContextFactory = documentContextFactory ?? throw new ArgumentNullException(nameof(documentContextFactory));
+    private readonly ILogger _logger = loggerFactory.CreateLogger<RazorDocumentMappingService>();
```

Comment:
are you relying on NRE to throw if `loggerFactory` is null?

Reply:
Technically, yes I suppose so, but in reality none of these will ever throw and my personal opinion would be to remove all of our argument validation on DI parameters. In MEF, if anything can't be injected there would be a CompositionFailedExtension and none of our code would load. In the language server, I imagine MS.Ext.DI would throw when we build the service collection, and the language server would fail to start.

Reply:
It's not a big deal here but overall, but in general I personally try to stay away from relying on NRE as much as possible. 
It makes it more difficult to debug compared to when we have proper exception handling (with proper message) and we know what to expect.

Reply:
I completely agree. My point here is that we do get a proper exception with a proper message from the various DI systems being used.